File: backend/app/schemas/venue.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime
from app.models.venue import MapType

class VenueBase(BaseModel):
    name: str = Field(..., min_length=1)
    address: str = Field(..., min_length=1)
    city: str = Field(..., min_length=1)
    state: str = Field(..., min_length=1)
    country: str = Field(..., min_length=1)
    latitude: float = Field(..., ge=-90.0, le=90.0)
    longitude: float = Field(..., ge=-180.0, le=180.0)
    map_type: MapType = MapType.GEOGRAPHIC

class VenueCreate(VenueBase):
    pass

class VenueUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1)
    address: Optional[str] = Field(None, min_length=1)
    city: Optional[str] = Field(None, min_length=1)
    state: Optional[str] = Field(None, min_length=1)
    country: Optional[str] = Field(None, min_length=1)
    latitude: Optional[float] = Field(None, ge=-90.0, le=90.0)
    longitude: Optional[float] = Field(None, ge=-180.0, le=180.0)
    map_type: Optional[MapType] = None

class VenueResponse(VenueBase):
    id: str
    created_by: str
    created_at: datetime
    updated_at: datetime

    model_config = {
        "from_attributes": True
    }

File: frontend/src/types/venue.ts
export interface FloorPlan {
  id: string;
  venueId: string;
  fileName: string;
  fileUrl: string;
  width: number;
  height: number;
  coordinateSystem: string;
  createdAt: string;
}

export interface GeoPoint {
  lat: number;
  lng: number;
}

export interface GeoFeature {
  id: string;
  name: string;
  coordinates: GeoPoint[];
  type: 'POLYGON' | 'POLYLINE' | 'POINT';
}

export interface Venue {
  id: string;
  name: string;
  address: string;
  city?: string;
  state?: string;
  country?: string;
  latitude?: number;
  longitude?: number;
  mapType?: 'GEOGRAPHIC' | 'FLOOR_PLAN';
  floorPlan?: FloorPlan;
  
  // Outdoor Map Configurable Geo-Features
  boundary?: GeoFeature;
  parkingLocations?: GeoFeature[];
  externalGates?: GeoFeature[];
  emergencyAccessRoutes?: GeoFeature[];
  nearbyRoads?: GeoFeature[];

  capacity: number;
  stats: {
    zones: number;
    gates: number;
    cameras: number;
    sensors: number;
  };
  createdAt?: string;
  updatedAt?: string;
}
